### ISSUE TITLE
multiple values per ldap attribute

### ISSUE THREAD

**Issue by otichy** [heiglandreas/authLdap#115]
Hi,

We have an ldap attribute that can have multiple values per person (it's a university ldap and teachers here can belong to more than one departments, students can be enrolled in multiple programs etc.]. Currently authldap only seems to evaluate the first of those values. 

Could the plugin load and evaluate more values per attribute?

Thanks!

Ondrej

**Comment by heiglandreas** [comment]
Can you give me an example?

**Comment by otichy** [comment]
The real life example is this. We have users (students), who have an attribute corresponding to the subject/programme they study and one student can study more subjects/programmes.

At this point, I have the filtera nd the group filter set to:
(&(uid=%s)(objectclass=cuniPerson))

and the group attribute to:
cuniStudySubject

I want to set a specific role to all students who study the subject/programme:
7310T033:Českýjazyk a literatura@ff.cuni.cz

which I have set under group-role mapping. As long as this is the only value the student has for this attribute, the role is set correctly. But students, who have multiple values of this attribute and the first one is not the one above, it does not work. So for a student who has the following values, the role is not set:

7310T270:Překladatelství: čeština - angličtina@ff.cuni.cz
7310T033:Český jazyk a literatura@ff.cuni.cz

Any ideas?

Thanks!

**Comment by heiglandreas** [comment]
Ah OK! I get the picture! I think I can fix that! Give me a day. I need to fix an error first (Please do NOT update to 1.4.20 when using ldaps!)

**Comment by heiglandreas** [comment]
The strange thing is, that it should already work… I've added some code to check for that and it shows that it actually should work.

Would you mind to enable debugging for authLdap? After a failing login there should be an entry ```LDAP groups:``` directly followed by ```Role from LDAP group``` (which might be empty).

The values will be in your php error-logfile.

That would help me getting to the cause of that a lot.

Thanks!

Andreas

**Comment by otichy** [comment]
Yes, it seems you are right, the log reports multiple values. It seems the problem may be in getting right the encoding of the string, because it works differently in the admin interface, in the string it reports to the log as set for the mapping and in the string from the ldap. 

E.g. in this case, the mapping string in the admin interface is se to 

"7310R033:Českýjazyk a literatura@ff.cuni.cz"

this is turned into:

"7310R033:\xc4\x8cesk\xc3\xbdjazyk a literatura@ff.cuni.cz"

in the log and is compared to:

"7310R033:\\u010cesk\\u00fd jazyk a literatura@ff.cuni.cz"

Which is derived from the ldap. So maybe I have just missed a space in the confusion of the codes and literals. I have originally thought the problem is in the multiple values, because while testing, I set the attribute to be loaded into the email field (to check what it looks like in the end). But only one of the multiple values is loaded into that field (which in hindsight is logical) and that in a modified form again - with the unicode characters displayed rendered, but with all spaces removed (maybe that happens in order to make it a valid email?).

I need to ask the students to try and log in now that I have spotted the missing space and I'll report if that was it.

Thanks again!
